src/include/sof/audio/format_generic.h

```diff
@@ -40,4 +40,14 @@ static inline int16_t sat_int16(int32_t x)
 		return (int16_t)x;
 }
 
+static inline int8_t sat_int8(int32_t x)
+{
+	if (x > INT8_MAX)
+		return INT8_MAX;
+	else if (x < INT8_MIN)
+		return INT8_MIN;
+	else
+		return (int8_t)x;
```

Comment:
all the `else`s can be dropped

Reply:
It returns int8 not uint8

Reply:
simply remove line 49, and keep line 50, @lyakh , do you mean this?

Reply:
```
	if (x > INT8_MAX)
		return INT8_MAX;
	if (x < INT8_MIN)
		return INT8_MIN;
	return (int8_t)x;
```